### PR TITLE
DTSPO-9746 - Remove acme

### PR DESCRIPTION
**Pull request by endakelly**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPO-9746

### Change description ###
Remove acme component
ACME component moved to https://github.com/hmcts/azure-enterprise-acme
Some resource moves and terraform import required
Adds Application.Read.All to bootstrap app reg to allow it to read ACME app reg in https://github.com/hmcts/azure-enterprise-acme

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: modules/subscription/service_principal.tf [hunk 1]
```diff
@@ -24,6 +24,16 @@ resource "azuread_application" "app" {
       id   = "06da0dbc-49e2-44d2-8312-53f166ab848a" # Directory.Read.All Delegated
       type = "Scope"
     }
+
+    resource_access {
+      id   = "5b567255-7703-4780-807c-7be8301ae99b" # Group.Read.All Application
```
[the remark this answers is not in the document]
Comment (endakelly) [reply modules/subscription/service_principal.tf:29]: It's for the acme pipeline. It reads groups using a data source and this is to enable that. It still needs to be granted admin consent afterwards. Also requires Application.ReadWrite.All to create the app registration.

I'm wondering is the acme module sort of redundant now as this is the only repo that uses it. Could avoid granting these permissions (which do seem excessive) and use the GA SP in a provider to do the more permissive things whilst using the service connection created here to create the non-Azure AD resources.

[the remark this answers is not in the document]
Comment (timja) [reply modules/subscription/service_principal.tf:29]: check if the module is used elsewhere but should be fine to inline.

why would it need to be in this repo?

ah the read is fine but not so keen on the readwrite one.

[the remark this answers is not in the document]
Comment (endakelly) [reply modules/subscription/service_principal.tf:29]: I meant in the azure-enterprise-acme repo.

GitHub's new search only shows terraform-module-acme-function referenced here, in the module repo and in ops-bootstrap.

[the remark this answers is not in the document]
Comment (timja) [reply modules/subscription/service_principal.tf:29]: what makes it required to move into the repo (not objecting)?

anyway sounds good as long as we don't give out readwrite.all

**Review by timja — APPROVED** [review]

**Pull request merged**